### ISSUE TITLE
[BUG] VSCode linter doesn't consider inherited fields

### ISSUE THREAD

**Issue by LilaRest** [zenstackhq/zenstack#1713]
**Description and expected behavior**
The VSCode linter seems to ignore that a model is inheriting some columns from another. In the below example, the `id` is inherited from an abstract model called `Model`, but the linter errors as if no `id` field was present at all.

**Screenshots**
![image](https://github.com/user-attachments/assets/485e91ef-8403-440f-9c21-a3f825ef5ca7)

**Environment (please complete the following information):**

-   ZenStack version: [e.g., 2.5.0]
-   Prisma version: [e.g., 5.7.0]
-   Database type: [e.g. Postgresql]

**Additional context**

**Comment by LilaRest** [comment]
And it blocks `zenstack generate` also, if that can help.
```
WARNING: Multiple versions of Zenstack packages detected. Run "zenstack info" to see details.
src/data/schema.zmodel:104:53 - Could not resolve reference to DataModelField named 'id'.
src/data/schema.zmodel:106:55 - Could not resolve reference to DataModelField named 'id'.
src/data/schema.zmodel:157:55 - Could not resolve reference to DataModelField named 'id'.
src/data/schema.zmodel:160:57 - Could not resolve reference to DataModelField named 'id'.
error: "zenstack" exited with code 1
```

**Comment by ymc9** [comment]
Hi @LilaRest , thanks for filing this. The `this` keyword was not properly resolved to the base model. This is already fixed in dev branch and will be part of v2.5.2.

However, your rule: `auth().membership?[orgId == this.id]` is not supported by ZenStack, as it involves comparing fields from different models (`Membership.orgId` and `Org.id`). It's a current limitation of ZenStack (inherited from Prisma's missing feature ...).

Maybe you can reverse the rule like `membership?[userId == auth().id]`, assuming `Org` has a `membership` relation field.

**Comment by ymc9** [comment]
Fixed in 2.6.0
